apps/wallet/src/shared/utils/bip39.test.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect } from 'vitest';
+
+import { normalizeMnemonics, validateMnemonics } from './bip39';
+
+describe('mnemonics', () => {
+    it('normalize mnemonics', () => {
+        expect(normalizeMnemonics(' Almost a Seed    Phrase')).toEqual(
+            'almost a seed phrase'
+        );
+    });
+
+    it('validate mnemonics', () => {
+        // Mnemonics length too short
+        expect(validateMnemonics('result')).toBe(false);
+
+        // Mnemonics length too long
+        expect(
+            validateMnemonics(
```

Comment:
Do we strictly support 12-word mnemonics or bigger acceptable as well?

Reply:
good point, we support 12-24 words mnemonics as per the underlying library: https://github.com/paulmillr/scure-bip39. added test vectors from official BIP32 spec.